homeassistant/components/snmp/switch.py

```diff
@@ -195,8 +197,36 @@ def __init__(
         self._command_payload_off = command_payload_off or payload_off
 
         self._state = None
-        self._payload_on = payload_on
-        self._payload_off = payload_off
+
+        try:
+            self._payload_on_int = Integer(payload_on)
+        except PyAsn1Error:
+            self._payload_on_int = None
+
+        try:
+            self._payload_off_int = Integer(payload_off)
+        except PyAsn1Error:
+            self._payload_off_int = None
+
+        try:
+            self._payload_on_regex = re.compile(payload_on)
```

Comment:
Is this needed for the fix? If not we shouldn't mix a feature addition with a bug fix.

Reply:
The addition of regex processing is not needed - it is the new feature.
It's such a small change that it seemed a waste to make two PRs when one would do.

Reply:
Please split the feature addition into a separate PR. Fixes are quicker to review and merge and can go into patch releases. That's not true for feature additions.

Reply:
Okay. Apologies.
I will move this to HACS for now

Reply:
Would you like me to withdraw the PR @MartinHjelmare ?

Reply:
I'd like this PR to only do the fix and not add the regex feature.

Reply:
@MartinHjelmare no problem - I will come back to this later in the year when I have time.
Hopefully, somebody will come across this and split it up to get SNMP switches working.